src/daemon/plugin.c

```diff
@@ -345,6 +349,7 @@ static int create_register_callback(llist_t **list, /* {{{ */
 
   cf = calloc(1, sizeof(*cf));
   if (cf == NULL) {
+    free_userdata(*ud);
```

Comment:
This dereferences `ud` without checking for `NULL`. I'd suggest to change `free_userdata` to take a `user_data_t*` (i.e. a pointer) instead and check for `NULL` there.